misc/config_tools/static_allocators/pio.py

```diff
@@ -47,28 +47,30 @@ def fn(board_etree, scenario_etree, allocation_etree):
         full = True
         pio_list = VALID_PIO
 
-    vuart_valid = ['ttyS0', 'ttyS1', 'ttyS2', 'ttyS3']
     hv_debug_console = lib.lib.parse_hv_console(scenario_etree)
 
     scenario_sos_vm_node = common.get_node("//vm[vm_type = 'SOS_VM']", scenario_etree)
     if scenario_sos_vm_node is not None:
         vm_id = common.get_node("./@id", scenario_sos_vm_node)
         if common.get_node("./legacy_vuart[@id = '0']/base/text()", scenario_sos_vm_node) != "INVALID_COM_BASE":
             vuart0_base = ""
-            if hv_debug_console in vuart_valid and hv_debug_console in native_ttys.keys() and native_ttys[hv_debug_console]['type'] == "portio":
-                vuart0_base = native_ttys[hv_debug_console]['base']
-                if vuart0_base in pio_list:
-                    remove_pio(pio_list, vuart0_base)
+            if hv_debug_console in native_ttys.keys():
+                if native_ttys[hv_debug_console]['type'] == "portio":
+                    vuart0_base = native_ttys[hv_debug_console]['base']
+                    if vuart0_base in pio_list:
+                        remove_pio(pio_list, vuart0_base)
+                else:
+                    vuart0_base = alloc_pio(pio_list)
+                    if full:
+                        logging.warning("All available pio bases are used by native fully. '{}' is taken by sos legacy vuart 0.".format(vuart0_base))
```

Comment:
```suggestion
                        logging.warning("All PIO bases available are in use. '{}' is taken by the Service VM legacy vuart0.".format(vuart0_base))
```

Does this new sentence reflect the situation correctly?

Reply:
looks fine with me. updated.